ports/broadcom/common-hal/busio/UART.c

```diff
@@ -208,7 +208,9 @@ void common_hal_busio_uart_construct(busio_uart_obj_t *self,
     self->sigint_enabled = sigint_enabled;
 
     if (rx != NULL) {
+        // Use the provided buffer when given.
         if (receiver_buffer != NULL) {
+            ringbuf_init(&self->ringbuf, receiver_buffer, receiver_buffer_size);
             self->ringbuf = (ringbuf_t) { receiver_buffer, receiver_buffer_size };
```

Comment:
looks funky

Reply:
Bad edit, and fixed another issue too. Thanks.